cmake/Tools/Platform/Android/android_post_build.py

```diff
@@ -0,0 +1,256 @@
+        raise AndroidPostBuildError(f"Unable to located the required 'engine_android.pak' file at location specified at {src_pak_file_full_path}. "
+                                    f"{PAK_FILE_INSTRUCTIONS}")
+
+    # Clear out the target folder first
+    safe_clear_folder(target_layout_root)
+
+    # Copy/Link the contents to the target folder
+    synchronize_folders(src_pak_file_full_path, target_layout_root)
+
+
+def apply_loose_layout(project_root: Path, target_layout_root: Path, android_app_root_path: Path, native_build_folder: str, build_config: str) -> None:
+    """
+    Apply the loose assets folder layout rules to the target assets folder
+
+    :param project_root:            The project folder root to look for the loose assets
+    :param target_layout_root:      The target layout destination of the loose assets
+    :param android_app_root_path:   The root of the 'app' project in the Gradle script
+    :param native_build_folder:     The sub-folder where the intermediate native project files are generated
+    :param build_config:            The build configuration used for the native build
+    :param is_monolithic:           Flag marking the deployment is for monolithic builds
+    """
+
+    android_cache_folder = project_root / 'Cache' / ASSET_PLATFORM_KEY
+    engine_json_marker = android_cache_folder / 'engine.json'
+    if not engine_json_marker.is_file():
+        raise AndroidPostBuildError(f"Assets have not been built for this project at ({project_root}) yet. "
+                                    f"Please run the AssetProcessor for this project first.")
+
+    # Clear out the target folder first
+    safe_clear_folder(target_layout_root)
+
+    # Copy/Link the contents to the target folder
+    synchronize_folders(android_cache_folder, target_layout_root)
+
+
+def post_build_action(android_app_root: Path, native_build_folder: str, build_config: str, project_root: Path, gradle_version: Version,
+                      asset_mode: str, asset_bundle_folder: str):
+    """
+    Perform the post-build logic for android native builds that will prepare the output folders by laying out the asset files
+    to their locations before the APK is generated.
+
+    :param android_app_root:    The root path of the 'app' project within the Android Gradle build script
+    :param native_build_folder: (For Gradle versions 7.x+) Gradle uses an intermediate sub-folder The under the 'app' project.
+    :param project_root:        The root of the project that the APK is being built for
+    :param gradle_version:      The version of Gradle used to build the APK (for validation)
+    :param build_config:        The native build configuration (Debug, Profile, Release)
+    :param asset_mode:          The desired asset mode to determine the layout rules
+    :param asset_bundle_folder: (For PAK asset modes) the location of where the PAK files are expected.
+    """
+
+    android_app_root_path = Path(android_app_root)
```

Comment:
nbd but it looks like `android_app_root` is already a `Path` type everywhere it is called.

Reply:
Fixed